test/e2e/util.go

```diff
@@ -1337,3 +1338,69 @@ func matchIPv4StringFamily(ipStrings []string) (string, error) {
 func matchIPv6StringFamily(ipStrings []string) (string, error) {
 	return util.MatchIPStringFamily(true /*ipv6*/, ipStrings)
 }
+
+// This is a replacement for e2epod.DeletePodWithWait(), which does not handle pods that
+// may be automatically restarted (https://issues.k8s.io/126785)
+func deletePodWithWait(ctx context.Context, c clientset.Interface, pod *v1.Pod) error {
+	if pod == nil {
+		return nil
+	}
+	if pod.UID == "" {
+		// We only recurse into deletePodWithWaitByName when UID is *not* set, to
+		// avoid infinite loops.
+		return deletePodWithWaitByName(ctx, c, pod.Name, pod.Namespace)
```

Comment:
this is kinda like providing it a second chance and then bailing if again there is no UID?

Reply:
No, it's because the caller might do something like:

```
pod := &v1.Pod{ ... }
_, err := client.CoreV1().Pods(namespace).Create(...)
...
deletePodWithWait(ctx, client, pod)
```

But since that pod object is just the one they created by hand before, it's not going to have the UID filled in. So in that case we just fall back to doing `deletePodWithWaitByName`, and fetch the pod details from the server.

I was trying to make both APIs still work in all scenarios just like they used to, but maybe it would make more sense to just say "you can only call this with a fully-filled-in Pod struct" and have it return an error if you break that rule?

Reply:
aaaaahhhh I see....jeez I kept thinking when could a pod come with no UID unless apiserver was sleeping, i forgot we did stupid things in test